tests/Feature/GroupBalancedDivisionTest.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+use App\Helpers\GroupBalancedDivision;
+use App\Models\Event;
+use App\Models\User;
+use App\Models\Course;
+use App\Models\Registration;
+use App\Models\Group;
+
+
+uses(\Illuminate\Foundation\Testing\RefreshDatabase::class);
+
+test('equal div only drinkers', function () {
+    $event = Event::factory()->create(['consider_alcohol' => true])->first();
+    $courses = Course::factory()->count(10)->create();
+    $groups = Group::factory()->count(10)->create(['event_id' => $event->id]);
+
+    foreach ($courses as $course) {
+        User::factory()->count(20)->create(['course_id' => $course->id]);
+    }
+
+    foreach (User::all() as $user) {
+        Registration::factory()->create([
```

Comment:
All attributes are set here, what added value does it offer to use a factory here instead of making a direct assignment? In my opinion, this should be implemented here without a factory and using direct assignment.